[kikokidayo] iP

kikokidayo opened the pull request:
# MomoBot

Momo helps you to make task management more intuitive, easy and fun. Juuust with a little _threatening_ twist 👹 ...



It contains numerous functionalities, the most important ones being:
1. Adding **to-do** items
2. Adding **events**
3. Adding **deadlines** 
4. **Listing** all tasks
5. **Searching** for tasks with a keyword
6. **Deleting** specific tasks
7. **Marking** tasks as complete/incomplete

So what are you waiting for! It's time to:
- Click onto this [link](https://github.com/kikokidayo/ip/releases/tag/A-Jar)
- Download the v0.1.0 jar file
- Start being productive!

Features coming soon:
- [ ] Cool stuff! 

Here's a little snippet of the code
```java
    public Momo(String filePath) {
        ui = new Ui();
        storage = new Storage(filePath);

        try {
            tasks = new TaskList(storage.load());

        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }

```

> "Download it now and your life will change forever" - Me

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kikokidayo committed:
Added greetings and renamed main class

kikokidayo committed:
Added logo customisation for the chatbot

kikokidayo committed:
Added echo

kikokidayo committed:
Debugged echo

kikokidayo committed:
Added support for list

kikokidayo committed:
Creating Task class extension file

kikokidayo committed:
Added a class for task

kikokidayo committed:
Add support for Mark and Unmark as done

kikokidayo committed:
Add basic implementation via Inheritance for ToDos, Events, Deadlines

kikokidayo committed:
Add support for different task types

kikokidayo committed:
Add automated Text UI Testing

kikokidayo committed:
Test input text

kikokidayo committed:
Deleting Windows testing file, renaming Duke in test file

kikokidayo committed:
Add exceptions and enums for all chatbot commands

kikokidayo committed:
Add delete support & Java collection support for Tasks list

kikokidayo committed:
Add basic file loading

kikokidayo committed:
Add to file string functionality for easier parsing

kikokidayo committed:
Add to file string parsing for ToDo

kikokidayo committed:
Update gitignore to include some output files

kikokidayo committed:
Add file syooirt

kikokidayo committed:
Add momo.txt output file

kikokidayo committed:
Change gitignore

kikokidayo committed:
Store dates as LocalDate objects in deadline and event

kikokidayo committed:
Improve formatting

kikokidayo committed:
Add support for dates on main program

kikokidayo committed:
Merge branch 'branch-Level-8'

kikokidayo committed:
Add Parser, Storage, Ui, and TaskList classes for improved OOP structure

kikokidayo committed:
Add CommandType enum to define supported command types

kikokidayo committed:
Add support for new OOP classes

kikokidayo committed:
Merge branch 'branch-A-MoreOOP'

kikokidayo committed:
Divide classes into packages

kikokidayo committed:
Merge branch 'branch-A-Packages'

kikokidayo committed:
Merge branch 'add-gradle-support'

kikokidayo committed:
Update package

kikokidayo committed:
Add Gradle support

kikokidayo committed:
Add more custom exceptions

kikokidayo committed:
Improve OOP by adding Command classes

kikokidayo committed:
Improve OOP further

kikokidayo committed:
Add JUnit test for Deadline task addition

kikokidayo committed:
Add stubs for TaskList and Storage for DeleteCommand test

kikokidayo committed:
Add test for DeleteCommand

kikokidayo committed:
Stop tracking momo.txt

kikokidayo committed:
Update gitignore

kikokidayo committed:
Fix bug to properly handle case where file is empty

kikokidayo committed:
Add JavaDocs for all files in Command package

kikokidayo committed:
Add JavaDocs for all files in Task package

kikokidayo committed:
Add JavaDocs for the rest of the files

kikokidayo committed:
Delete unnecessary importys

kikokidayo committed:
Delete unnecessary imports on Command

kikokidayo committed:
Correcting try-catch formatting

kikokidayo committed:
Correct if-else format

kikokidayo committed:
Add new FindCommand

kikokidayo committed:
Add Find CommandType

kikokidayo committed:
Add support for Find command in main control flow

kikokidayo committed:
Add getTask() method in Task for Find command

kikokidayo committed:
Add parsing for "find" input in Parser

kikokidayo committed:
Merge branch 'branch-A-JavaDoc'

kikokidayo committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/momo/Ui.java
#	src/main/java/momo/command/Command.java
#	src/main/java/momo/task/TaskList.java

kikokidayo committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/momo/Momo.java
#	src/main/java/momo/task/Task.java

kikokidayo committed:
Add line wrapping after 120 characters for all files

Virusrwj223 commented on src/main/java/momo/Ui.java:
```diff
@@ -0,0 +1,85 @@
+ * Handles Ui related functionality in the chatbot program, handling some user interactions
+ */
+public class Ui {
+    public static void showHorizontalLine() {
```
standardise printing function names to be prefixed by either "print" or show

kikokidayo replied:
> standardise printing function names to be prefixed by either "print" or show

Sorry, may I clarify where this is stated to be the convention in the Coding Standard document? Can't seem to find it

Virusrwj223 commented on src/main/java/momo/command/AddCommand.java:
```diff
@@ -0,0 +1,43 @@
+ * to the storage and for printing confirmation messages that display tasks
+ * which have been added
+ */
+public class AddCommand extends Command {
```
Consider using a descriptive name. The name "AddCommand" could be mistaken for mathematical addition. Maybe consider using "AddTaskCommand"?

kikokidayo replied:
Good point! I also realised it was an issue because the coding quality section addresses how class names should be nouns not verbs. So I think I might change it to "CommandAdder" instead

kikokidayo replied:
> Consider using a descriptive name. The name "AddCommand" could be mistaken for mathematical addition. Maybe consider using "AddTaskCommand"?

Good point! I also realised it was an issue because the coding quality section addresses how class names should be nouns not verbs. So I think I might change it to "CommandAdder" instead

Virusrwj223 commented on src/main/java/momo/command/EventCommand.java:
```diff
@@ -0,0 +1,55 @@
+    public static void run(String input, Storage storage, TaskList tasks) throws InvalidCommandException,
+            StorageException {
+        try {
+            String desc = input.substring(5).trim();
```
Consider changing the name from shorthand to its full name (i.e description). Applies to other files as well.

Virusrwj223 commented on src/main/java/momo/task/Task.java:
```diff
@@ -0,0 +1,44 @@
+    String task;
+    boolean isComplete;
+
+    public Task(String task, boolean isComplete) {
```
Consider renaming the task parameter to something more specific so that it is not confused with the Task class.

Virusrwj223 commented on src/main/java/momo/task/Task.java:
```diff
@@ -0,0 +1,44 @@
+    /**
+     * Marks the task object as complete
+     */
+    public void markComplete() {
```
May want to consider standardisation with "unmark" (i.e "mark" in this case)?

kikokidayo replied:
> May want to consider standardisation with "unmark" (i.e "mark" in this case)?

True, will look into it

Virusrwj223 reviewed (commented):
looks good 👍

kikokidayo committed:
Add CheckStyle config files

kikokidayo committed:
Move exceptions to package

kikokidayo committed:
Add JavaFX files

kikokidayo committed:
Add greet ability on GUI

kikokidayo committed:
Merge branch 'branch-Level-10'

kikokidayo committed:
Merge branch 'master' into branch-Level-10

kikokidayo committed:
Add support to Ui for scanning and printing JavaFx dialogue

kikokidayo committed:
Edit MainWindow setup

kikokidayo committed:
Add processCommand function to process GUI commands

kikokidayo committed:
Add GUI support for Find command

kikokidayo committed:
Add GUI print dialogue support for printing tasks added

kikokidayo committed:
Add GUI support for adding any type of task

kikokidayo committed:
Add GUI support for delete task command

kikokidayo committed:
Add GUI support for marking and unmarking tasks

kikokidayo committed:
Merge branch 'branch-Level-10'

tohjh commented on src/main/java/momo/task/TaskList.java:
```diff
@@ -0,0 +1,115 @@
+    }
+
+    public Task getTask(int index) {
+
```
extra line out of nowhere, might be nicer looking to remove it

tohjh commented on src/main/java/momo/task/TaskList.java:
```diff
@@ -0,0 +1,115 @@
+
+            // For deadline and events assumption is that file will always be correctly formatted
+            switch (inputs[0]) {
+                case "T" -> {
```
if i am not wrong the various cases shouldn't be indented based on the guide

tohjh commented on src/main/java/momo/task/Event.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    public String toString() {
+        return "[E]" + super.toString() + "(" + from.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + " to " + to.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```
nice breaking up of the long line :)

tohjh commented on src/test/java/momo/task/DeadlineTest.java:
```diff
@@ -0,0 +1,26 @@
+    @Test
+    void toStringTest() {
+        // Check if task list string is formatted properly
+        assertEquals("[D][ ] return book (Apr 5 2021)", new Deadline("return book", LocalDate.parse("2021-04-05"), false).toString(), "Deadline string should be formatted properly for task list display");
```
line seem abit long, might be better to break it up

tohjh commented on src/main/java/momo/command/TodoCommand.java:
```diff
@@ -0,0 +1,34 @@
+        String desc = input.substring(4).trim();
+
+        if (desc.isEmpty()) {
+            throw new InvalidCommandException("Where's the description of your task?!");
```
i love the tone of your chatbot!!

tohjh reviewed (commented):
Some minor formatting and comments

kikokidayo committed:
Configure gradle to enable assertions

kikokidayo committed:
Add assertions for text is not empty & command is not null in Ui class

kikokidayo committed:
Add Assertions to the the children of AddCommands in the Command class

The run methods in most AddCommand classes contain task,tasks and UI
operations that must maintain certain invariants. If there are no
assertions, tasks and UI being null could go unnoticed and be processed
causing errors.

Let's add assertions to ensure tasks, task and UI is never null

Using such assertions helps developers catch errors early, ensuring the
proper initialisation of objects before being used.

kikokidayo committed:
Add another assertion to AddCommand

kikokidayo committed:
Add assertions to more classes

kikokidayo committed:
Replace magic number 6 with named constant COMMAND_PREFIX_OFFSET

kikokidayo committed:
Extract getTask method & replace magic number with GET_COMMAND_PREFIX

The logic for extracting the task index and parsing commands
includes a magic number (8), which reduces readability. The task
extraction logic is inlined, making the code harder to follow and reuse,
violating SLAP.

Let's,
* replace magic number 8 with the constant GET_COMMAND_PREFIX to
clarify its role in extracting substring after the command prefix
* extract the getTask() method to avoid multiple levels of abstraction
in the code

kikokidayo committed:
Fix coding style

kikokidayo committed:
Fix import order

kikokidayo committed:
Fix import order & replace command prefix magic number with constant

kikokidayo committed:
Extract getTask() method to adhere to SLAP and fix import method

kikokidayo committed:
Get rid of magic number

kikokidayo committed:
Correct swtich statement styling

kikokidayo committed:
Improve code quality

kikokidayo committed:
Merge pull request #1 from kikokidayo/branch-A-Assertions

Branch-A-Assertions

kikokidayo committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/momo/command/DeleteCommand.java
#	src/main/java/momo/command/EventCommand.java

kikokidayo committed:
Merge pull request #2 from kikokidayo/branch-A-CodeQuality

Branch-A-CodeQuality

kikokidayo committed:
Set up CI

kikokidayo committed:
Add new archive command class for the extension

kikokidayo committed:
Add archive command type in the command enum

kikokidayo committed:
Add archive command support into parser and process command method

kikokidayo committed:
Handle new archive.txt file creation in storage

kikokidayo committed:
Update .gitignore to ignore archive text file

kikokidayo committed:
Improve code quality and Momo chatbot personality

kikokidayo committed:
Improve code quality, add more javadocs

kikokidayo committed:
Add Ai acknowledgement

kikokidayo committed:
Add AI acknowledgement

kikokidayo committed:
Enhance GUI of momo chatbot

kikokidayo committed:
Improve error message

kikokidayo committed:
Add JUnit tests for mark and unmark command

kikokidayo committed:
Update exception message to include all Momo features

kikokidayo committed:
Update README and screenshot of the Momo chatbot GUI

kikokidayo committed:
Update README

kikokidayo committed:
Edit README.md

kikokidayo committed:
Update runtest.sh for CI

kikokidayo committed:
Update gradle.yml file for CI

kikokidayo committed:
Fix header comments
